projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/interop/LLVMTckTest.java

```diff
@@ -0,0 +1,410 @@
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.truffle.llvm.test.interop;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+import com.oracle.truffle.api.source.Source;
+import com.oracle.truffle.api.vm.PolyglotEngine;
+import com.oracle.truffle.api.vm.PolyglotEngine.Builder;
+import com.oracle.truffle.llvm.nodes.impl.base.LLVMLanguage;
+import com.oracle.truffle.llvm.test.LLVMPaths;
+import com.oracle.truffle.llvm.tools.Clang;
+import com.oracle.truffle.llvm.tools.Clang.ClangOptions;
+import com.oracle.truffle.llvm.tools.Opt;
+import com.oracle.truffle.llvm.tools.Opt.OptOptions;
+import com.oracle.truffle.llvm.tools.Opt.OptOptions.Pass;
+import com.oracle.truffle.tck.TruffleTCK;
+
+public class LLVMTckTest extends TruffleTCK {
+    private static final String FILENAME = "tck";
+
+    @Test
+    public void testVerifyPresence() {
+        PolyglotEngine vm = PolyglotEngine.newBuilder().build();
+        assertTrue("Our language is present", vm.getLanguages().containsKey(LLVMLanguage.LLVM_IR_MIME_TYPE));
+        vm.dispose();
+    }
+
+    @Override
+    protected PolyglotEngine prepareVM(Builder builder) throws Exception {
+        PolyglotEngine engine = builder.build();
+        try {
+            File cFile = new File(LLVMPaths.INTEROP_TESTS, FILENAME + ".c");
+            File bcFile = File.createTempFile(LLVMPaths.INTEROP_TESTS + "/" + "bc_" + FILENAME, ".ll");
+            File bcOptFile = File.createTempFile(LLVMPaths.INTEROP_TESTS + "/" + "bcopt_" + FILENAME, ".ll");
+            Clang.compileToLLVMIR(cFile, bcFile, ClangOptions.builder());
+            Opt.optimizeBitcodeFile(bcFile, bcOptFile, OptOptions.builder().pass(Pass.MEM_TO_REG));
+            engine.eval(Source.newBuilder(bcOptFile).build()).as(Integer.class);
+        } catch (IOException e) {
+            throw new AssertionError(e);
+        }
+        return engine;
+    }
+
+    @Override
+    protected String mimeType() {
+        return LLVMLanguage.LLVM_IR_MIME_TYPE;
+    }
+
+    @Override
+    protected String fourtyTwo() {
+        return "fourtyTwo";
+    }
+
+    @Override
+    protected String identity() {
+        return "identity";
+    }
+
+    @Override
+    protected String plus(Class<?> type1, Class<?> type2) {
+        return "plus";
+    }
+
+    @Override
+    protected String returnsNull() {
+        return "returnsNull";
+    }
+
+    @Override
+    protected String applyNumbers() {
+        return "apply";
+    }
+
+    @Override
+    protected String compoundObject() {
+        return "compoundObject";
+    }
+
+    @Override
+    protected String valuesObject() {
+        return "valuesObject";
+    }
+
+    @Override
+    protected String invalidCode() {
+        // @formatter:off
+        return
+            "f unction main() {\n" +
+            "  retu rn 42;\n" +
+            "}\n";
+        // @formatter:on
```

Comment:
Maybe, but then one single test case is not sufficient. I can specify "invalid" IR which results in parse errors, I can specify "invalid" IR which parses without errors but produces runtime errors when the parsed code is executed, and I can specify something which is not IR at all, like I did in the beginning. And I think all of those cases should be checked. But I also think that this has more to do with the parser itself, than with interop …